sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -583,6 +689,10 @@ class CudfRegexTranspiler(mode: RegexMode) {
               // Java, it's treated as a single (b$ and b$$ are synonymous), so we create 
               // an empty RegexAST that outputs to empty string
               RegexEmpty()
+            case Some(RegexChar(ch)) if mode == RegexReplaceMode
+                && lineTerminatorChars.contains(ch) =>
+                throw new RegexUnsupportedException("Regex sequences with a line terminator " 
```

Comment:
So this is a strange edge case where the behavior in Java is a little strange. Here is an example:

Input: `AB\r\r\r`
Regex: `\r$`
Replacement: "_REPLACE_"

In Java, this becomes `AB\r_REPLACE__REPLACE_` (so it randomly behaves in a multiline mode), in cuDF, this becomes `AB\r\r_REPLACE_` (which is the same in Python). Since these are anchors, the underlying regex engine needs to be modified to handle this case somewhat properly, and from a cuDF perspective, this is not really defined behavior, so it wouldn't make any sense to fix it on that side.